grammar.js

```diff
@@ -384,61 +384,77 @@ module.exports = grammar({
 
     keyword_array: _ => make_keyword("array"), // not included in _type since it's a constructor literal
 
-    _type: $ => choice(
-      $.keyword_boolean,
-      $.bit,
-      $.keyword_binary,
-
-      $.keyword_smallserial,
-      $.keyword_serial,
-      $.keyword_bigserial,
-
-      $.tinyint,
-      $.smallint,
-      $.mediumint,
-      $.int,
-      $.bigint,
-      $.decimal,
-      $.numeric,
-      $.double,
-      $.float,
-
-      $.keyword_money,
-
-      $.char,
-      $.varchar,
-      $.numeric,
-      $.keyword_string,
-      $.keyword_text,
-
-      $.keyword_uuid,
-
-      $.keyword_json,
-      $.keyword_jsonb,
-      $.keyword_xml,
-
-      $.keyword_bytea,
-      $.keyword_inet,
-
-      $.enum,
-
-      $.keyword_date,
-      $.keyword_datetime,
-      $.keyword_timestamp,
-      $.keyword_timestamptz,
-      $.keyword_interval,
-
-      $.keyword_geometry,
-      $.keyword_geography,
-      $.keyword_box2d,
-      $.keyword_box3d,
-
-      $.keyword_oid,
-      $.keyword_name,
-      $.keyword_regclass,
-      $.keyword_regnamespace,
-      $.keyword_regproc,
-      $.keyword_regtype,
+    _type: $ => seq(
+      choice(
+        $.keyword_boolean,
+        $.bit,
+        $.keyword_binary,
+
+        $.keyword_smallserial,
+        $.keyword_serial,
+        $.keyword_bigserial,
+
+        $.tinyint,
+        $.smallint,
+        $.mediumint,
+        $.int,
+        $.bigint,
+        $.decimal,
+        $.numeric,
+        $.double,
+        $.float,
+
+        $.keyword_money,
+
+        $.char,
+        $.varchar,
+        $.numeric,
+        $.keyword_string,
+        $.keyword_text,
+
+        $.keyword_uuid,
+
+        $.keyword_json,
+        $.keyword_jsonb,
+        $.keyword_xml,
+
+        $.keyword_bytea,
+        $.keyword_inet,
+
+        $.enum,
+
+        $.keyword_date,
+        $.keyword_datetime,
+        $.keyword_timestamp,
+        $.keyword_timestamptz,
+        $.keyword_interval,
+
+        $.keyword_geometry,
+        $.keyword_geography,
+        $.keyword_box2d,
+        $.keyword_box3d,
+
+        $.keyword_oid,
+        $.keyword_name,
+        $.keyword_regclass,
+        $.keyword_regnamespace,
+        $.keyword_regproc,
+        $.keyword_regtype,
+      ),
+      optional($.array_size_definition)
+    ),
+
+    array_size_definition: $ => seq(
+      choice(
+        seq(optional($.keyword_array), $._array_size_definition),
+        seq( $._array_size_definition, $._array_size_definition),
+        $.keyword_array,
+      ),
+    ),
+    _array_size_definition: $ => seq(
+      '[',
+      optional(field("size", alias($._integer, $.literal))),
+      ']'
```

Comment:
```suggestion
      seq(
        optional($.keyword_array),
        repeat(
          seq(
            '[',
            optional(field("size", alias($._integer, $.literal))),
            ']'
          )
        )
      ),
```

Postgres allows n-dimensional arrays > 2; `repeat` is 0 or more times so this should cover all possibilities

Reply:
the ARRAY keyword works only for 1D arrays
shouldn't it be `repeat1()` instead of `repeat()`?

Reply:
I have pushed a version, which allows for more dimensions. Unfortunately, it requires a conflict rule.

Reply:
`repeat` would allow `array`, `array[n]`, `array[n][n]....` and cover all three at once (including the zero-repeats case of just the keyword), but it is more permissive than necessary if the keyword is only used with 1d arrays. I'm fine either way though.

if you replace the top and bottom options in your latest with `seq($.keyword_array, optional($._array_size_definition))` do you still need the conflict rule?

Reply:
Ah yes, this resolves the conflict.

I have pushed it to my branch, but for some reason it does not show up in the PR :confused: